src/codecTools/decodeCodec.ts

```diff
@@ -48,26 +56,29 @@ export function decodeCodec<TCodec extends t.Any>(
   codec: TCodec,
   txt: string | object | object[] | null | unknown,
   parse = true,
-  // Warning: DO NOT LOG SENSITIVE DATA WITH THIS!
-  customErrorFromContext: (validationContext: t.Context) => string = () => '',
+  customErrorFromContext:
+    | ((validationContext: t.Context) => string)
+    | undefined = undefined,
 ): t.TypeOf<TCodec> {
   // uncomment the below to view the response pre-decode
   const decoded = codec.decode(
     parse && typeof txt === 'string' ? JSON.parse(txt) : txt,
   );
   // Log errors on failure
-  // eslint-disable-next-line no-underscore-dangle
-  if (decoded._tag === 'Left') {
+  if (isLeft(decoded)) {
```

Comment:
What does `isLeft(decoded)` mean?

Reply:
It's an io-ts thing, where "Left" is used for errors, and "Right" is used for success values. If a response is "Left" it's typically an error, and this helper function provided by the lib just checks to see if the response is an error